www/include/eventLogs/xml/data.php

```diff
@@ -360,13 +365,15 @@
 
 $whereOutput = "";
 if (isset($output) && $output != "") {
-    $whereOutput = " AND logs.output like '%" . $pearDBO->escape($output) . "%' ";
+    $whereOutput = " AND logs.output like '%" . CentreonUtils::escapeSecure($output) . "%' ";
 }
 
 $innerJoinEngineLog = "";
 if ($engine == "true" && isset($openid) && $openid != "") {
-    $innerJoinEngineLog = " INNER JOIN instances i ON i.name = logs.instance_name AND i.instance_id IN (" .
-        $pearDBO->escape($openid) . ") ";
+    if (preg_match('/([0-9]+)?(,[0-9]+)*/', $openid, $matches)) {
+        $queryValues['openid'] = [\PDO::PARAM_STR => CentreonUtils::escapeSecure($openid)];
+        $innerJoinEngineLog = " INNER JOIN instances i ON i.name = logs.instance_name AND i.instance_id IN ( :openid );";
```

Comment:
you've sanitized $openId as a string. but if $openId = '1,-1,"toto", ";",  12'. Won't the query break ?
The line 373 will return true, but $matches will be empty. Resulting in an injectable $openId